### DISCUSSION TITLE
Unable to Web GUI after upgrading 3.5.3

### DISCUSSION THREAD

**Discussion by chanukad24** [netbox-community/netbox#12856] [Help Wanted!]
Hi Guys,

I am unable to open the web GUI after I upgrade netbox from 3.4.5 to 3.5.3.  Netbox service is running and no errors are displayed. I am not sure what else to check. pls help. 

![image](https://github.com/netbox-community/netbox/assets/136179008/c7b66443-1571-4996-9012-3f536efb98e3)
![image](https://github.com/netbox-community/netbox/assets/136179008/b67a3274-9bdf-4d23-acee-b0cb04ea2dc4)

**Comment by peterbaumert** [comment]
What does the browser show and is nginx running?

**Reply by chanukad24** [reply]
Browser says site can't be reached. Yes Nginx is running. 
![image](https://github.com/netbox-community/netbox/assets/136179008/b7326f87-4b96-4eb5-aee0-7a76deb0ff86)

**Reply by chanukad24** [reply]
I just noticed there is an error says "Unit gunicorn.service could not be found." when I run "sudo systemctl status gunicorn". Not sure could this be the problem or not.

**Comment by candlerb** [comment]
> Browser says site can't be reached

Please can you show the *exact* error that you see? Screenshot the browser if necessary.

Don't worry about "systemctl status gunicorn". It probably means you have an extra copy of gunicorn installed globally, but Netbox won't use that.

**Reply by chanukad24** [reply]
This is what browser displays. 
![image](https://github.com/netbox-community/netbox/assets/136179008/cac402b6-e6ae-4dbc-9d56-19a0e77cb1c5)

**Reply by chanukad24** [reply]
I have a domain set. I can ping to domain name. no issues. Reachability is fine to the server.

**Reply by candlerb** [reply]
OK, so the error is "connection timed out". This means two possibilities:
* there is no network reachability between your browser and the server (so the TCP connection timed out while it was trying to establish). If you can ping it, then it could be firewalls blocking the TCP port.
* the web server received the connection, but took too long to respond

You may get a clearer response by checking `time curl -v https://netbox.your.domain/`. This will show you if the connection establishes, if TLS establishes, and if it gets as far as sending request with headers.  Also, `time` will show the actual duration before it gives up.

There is one thing I can think of which might have changed in v3.5, and that's the dashboard.  The dashboard widgets now make *outbound* connections from the server to the Internet; if your server is behind a firewall that doesn't allow outbound connections, they may time out. However, it's supposed to allow only a few seconds for the attempt.

**Reply by chanukad24** [reply]
Hi below is the output. I have allowed netbox to connect to internet as well. Still unable to load GUI.

* Connected to netbox.your.domain (X.X.X.X) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS header, Finished (20):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.2 (OUT), TLS header, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=netbox.your.domain
*  start date: Jun  9 03:40:41 2023 GMT
*  expire date: Sep  7 03:40:40 2023 GMT
*  subjectAltName: host "netbox.your.domain" matched cert's "netbox.your.domain"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
> GET / HTTP/1.1
> Host: netbox.your.domain
> User-Agent: curl/7.81.0
> Accept: */*
>
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Server: nginx/1.18.0 (Ubuntu)
< Date: Sun, 11 Jun 2023 08:30:19 GMT
< Content-Type: text/html; charset=utf-8
< Content-Length: 27621
< Connection: keep-alive
< X-Request-ID: 37bd494f-f0fb-42bd-aa9f-bfb30de18cfd
< X-Content-Type-Options: nosniff
< Referrer-Policy: same-origin
< Cross-Origin-Opener-Policy: same-origin
< X-Frame-Options: SAMEORIGIN
< Vary: Cookie, origin
< Set-Cookie: csrftoken=gyAMfygobW3pkWgftsrNCwBFY7Ethcmcw; expires=Sun, 09 Jun 2024 08:30:19 GMT; Max-Age=31449600; Path=/; SameSite=Lax
<

**Reply by candlerb** [reply]
That's not the complete output of the command I gave, as it doesn't include the return string from `time`. You may have to wait several minutes, and there could be some additional data after the headers sent when it times out.

**Reply by chanukad24** [reply]
<!DOCTYPE html>
<html
  lang="en"
  data-netbox-url-name="home"
  data-netbox-base-path=""


      data-netbox-color-mode="unset"


  >
  <head>
    <meta charset="UTF-8" />
    <meta
      name="viewport"
      content="initial-scale=1, maximum-scale=1, user-scalable=no, width=device-width, viewport-fit=cover"
    />


    <title>Home | NetBox</title>

    <script
      type="text/javascript"
      src="/static/setmode.js"
      onerror="window.location='/media-failure/?filename=setmode.js'">
    </script>

    <script type="text/javascript">
      (function () {
          initMode()
      })();
      window.CSRF_TOKEN = "gtQiKSAkgdTUtXieHipRCZP6TaCpZDwqmRgUS49gueFNI74kvXGu4lgBH76L1PyM";
    </script>


    <link
      rel="stylesheet"
      href="/static/netbox-external.css?v=3.5.3"
      onerror="window.location='/media-failure/?filename=netbox-external.css'"
    />
    <link
      rel="stylesheet"
      href="/static/netbox-light.css?v=3.5.3"
      onerror="window.location='/media-failure/?filename=netbox-light.css'"
    />
    <link
      rel="stylesheet"
      href="/static/netbox-dark.css?v=3.5.3"
      onerror="window.location='/media-failure/?filename=netbox-dark.css'"
    />
    <link
      rel="stylesheet"
      media="print"
      href="/static/netbox-print.css?v=3.5.3"
      onerror="window.location='/media-failure/?filename=netbox-print.css'"
    />
    <link rel="icon" type="image/png" href="/static/netbox.ico" />
    <link rel="apple-touch-icon" type="image/png" href="/static/netbox_touch-icon-180.png" />


    <script
      type="text/javascript"
      src="/static/netbox.js?v=3.5.3"
      onerror="window.location='/media-failure/?filename=netbox.js'">
    </script>



  </head>

  <body>
    <script type="text/javascript">
      function checkSideNav() {
        // Check localStorage to see if the sidebar should be pinned.
        var sideNavRaw = localStorage.getItem('netbox-sidenav');
        // Determine if the device has a small screeen. This media query is equivalent to
        // bootstrap's media-breakpoint-down(lg) breakpoint mixin, which is what the sidenav's
        // CSS uses.
        var isSmallScreen = window.matchMedia('(max-width: 991.98px)').matches;
        if (typeof sideNavRaw === 'string') {
          var sideNavState = JSON.parse(sideNavRaw);
          if (sideNavState.pinned === true && !isSmallScreen) {
            // If the sidebar should be pinned and this is not a small screen, set the appropriate
            // body attributes prior to the rest of the content rendering. This prevents
            // jumpy/glitchy behavior on page reloads.
            document.body.setAttribute('data-sidenav-pinned', '');
            document.body.setAttribute('data-sidenav-show', '');
            document.body.removeAttribute('data-sidenav-hidden');
          } else {
            document.body.removeAttribute('data-sidenav-pinned');
            document.body.setAttribute('data-sidenav-hidden', '');
          }
        }
      }
      window.addEventListener('resize', function(){ checkSideNav() });
      checkSideNav();
    </script>




  <div class="container-fluid px-0">
    <main class="layout">




<nav class="sidenav noprint" id="sidenav" data-simplebar>
  <div class="sidenav-header">




    <a class="sidenav-brand" href="/">
      <img src="/static/netbox_logo.svg" height="48" class="sidenav-brand-img" alt="NetBox Logo">
    </a>


    <a class="sidenav-brand-icon" href="/">
      <img src="/static/netbox_icon.svg" height="32" class="sidenav-brand-img" alt="NetBox Logo">
    </a>


    <button class="btn btn-sm btn-ghost-primary sidenav-toggle">
      <div class="sidenav-toggle-icon">
        <i class="mdi mdi-pin"></i>
      </div>
    </button>

  </div>

  <div class="sidenav-inner h-100 mb-auto">


    <div class="collapse sidenav-collapse pb-4">




<ul class="navbar-nav">

</ul>


    </div>


    <div class="profile-button-container">

  <div class="btn-group">
    <a class="btn btn-primary ws-nowrap" type="button" href="/login/?next=/">
      <i class="mdi mdi-login-variant"></i> Log In
    </a>
    <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split" data-bs-toggle="dropdown">
      <span class="visually-hidden">Toggle Dropdown</span>
    </button>
    <ul class="dropdown-menu dropdown-menu-end">
      <li>
        <button class="dropdown-item color-mode-toggle">
          <i class="color-mode-icon mdi mdi-lightbulb"></i>
          <span class="color-mode-text">Dark Mode</span>
        </button>
      </li>
    </ul>
  </div>


    </div>
  </div>
</nav>



      <div class="content-container" tabindex="-2">


        <div class="p-2 printonly">
          <img src="/static/netbox_logo.svg" alt="NetBox logo" width="200px" />
        </div>


        <nav class="navbar navbar-light sticky-top flex-md-nowrap p-1 mb-3 search container-fluid noprint">


            <div class="nav-mobile">
              <div class="nav-mobile-top">
                <a class="sidebar-logo p-2 d-block" href="/">
                  <img src="/static/netbox_logo.svg" alt="NetBox logo" height="40" />
                </a>
                <button type="button" aria-label="Toggle Navigation" class="navbar-toggler sidenav-toggle-mobile">
                  <span class="navbar-toggler-icon"></span>
                </button>
              </div>
              <div class="d-flex my-1 flex-grow-1 justify-content-center w-100">
                <form class="input-group" action="/search/" method="get">
  <input name="q" type="text" aria-label="Search" placeholder="Search" class="form-control" />
  <button class="btn btn-primary" type="submit">
    <i class="mdi mdi-magnify"></i>
  </button>
</form>

              </div>
            </div>


            <div class="row search-container">


              <div class="col-3 d-flex flex-grow-1 ps-0"></div>


              <div class="col-6 d-flex flex-grow-1 justify-content-center">
                <form class="input-group" action="/search/" method="get">
  <input name="q" type="text" aria-label="Search" placeholder="Search" class="form-control" />
  <button class="btn btn-primary" type="submit">
    <i class="mdi mdi-magnify"></i>
  </button>
</form>

              </div>


              <div class="col-3 d-flex flex-grow-1 pe-0 justify-content-end">

  <div class="btn-group">
    <a class="btn btn-primary ws-nowrap" type="button" href="/login/?next=/">
      <i class="mdi mdi-login-variant"></i> Log In
    </a>
    <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split" data-bs-toggle="dropdown">
      <span class="visually-hidden">Toggle Dropdown</span>
    </button>
    <ul class="dropdown-menu dropdown-menu-end">
      <li>
        <button class="dropdown-item color-mode-toggle">
          <i class="color-mode-icon mdi mdi-lightbulb"></i>
          <span class="color-mode-text">Dark Mode</span>
        </button>
      </li>
    </ul>
  </div>


              </div>

            </div>

        </nav>













        <div id="content" class="container-fluid content px-0 m-0">



  <div class="grid-stack" id="dashboard">



<div
    class="grid-stack-item"
    gs-w="4"
    gs-h="2"
    gs-x="None"
    gs-y="None"
    gs-id="e2fe4b28-c24a-4105-a446-77cbee918025"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-secondary p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/e2fe4b28-c24a-4105-a446-77cbee918025/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/e2fe4b28-c24a-4105-a446-77cbee918025/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>Organization</strong>

    </div>
    <div class="card-body p-2 overflow-auto">

  <div class="list-group list-group-flush">


        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Sites
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Tenants
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Contacts
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>


  </div>


    </div>
  </div>
</div>




<div
    class="grid-stack-item"
    gs-w="4"
    gs-h="3"
    gs-x="None"
    gs-y="None"
    gs-id="ee2a8edc-f68d-49b0-bc1e-09dc6415dcef"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-secondary p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/ee2a8edc-f68d-49b0-bc1e-09dc6415dcef/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/ee2a8edc-f68d-49b0-bc1e-09dc6415dcef/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>IPAM</strong>

    </div>
    <div class="card-body p-2 overflow-auto">

  <div class="list-group list-group-flush">


        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          VRFs
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Aggregates
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Prefixes
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          IP Ranges
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          IP Addresses
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          VLANs
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>


  </div>


    </div>
  </div>
</div>




<div
    class="grid-stack-item"
    gs-w="4"
    gs-h="2"
    gs-x="None"
    gs-y="None"
    gs-id="db435239-bd08-40c8-86e8-9d7c4b2c4a7d"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-green p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/db435239-bd08-40c8-86e8-9d7c4b2c4a7d/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/db435239-bd08-40c8-86e8-9d7c4b2c4a7d/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>Welcome!</strong>

    </div>
    <div class="card-body p-2 overflow-auto">
      <div class="rendered-markdown"><p>This is your personal dashboard. Feel free to customize it by rearranging, resizing, or removing widgets. You can also add new widgets using the "add widget" button below. Any changes affect only <em>your</em> dashboard, so feel free to experiment!</p></div>
    </div>
  </div>
</div>




<div
    class="grid-stack-item"
    gs-w="4"
    gs-h="2"
    gs-x="None"
    gs-y="None"
    gs-id="351c8db0-0c87-4876-809e-dbf8b1eeede0"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-secondary p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/351c8db0-0c87-4876-809e-dbf8b1eeede0/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/351c8db0-0c87-4876-809e-dbf8b1eeede0/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>Circuits</strong>

    </div>
    <div class="card-body p-2 overflow-auto">

  <div class="list-group list-group-flush">


        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Providers
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Circuits
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Provider Networks
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>


  </div>


    </div>
  </div>
</div>




<div
    class="grid-stack-item"
    gs-w="4"
    gs-h="4"
    gs-x="None"
    gs-y="None"
    gs-id="9c71af50-dbc6-42f9-9803-b0f971078c62"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-secondary p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/9c71af50-dbc6-42f9-9803-b0f971078c62/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
* TLSv1.2 (IN), TLS header, Supplemental data (23):
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/9c71af50-dbc6-42f9-9803-b0f971078c62/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>NetBox News</strong>

    </div>
    <div class="card-body p-2 overflow-auto">

  <div class="list-group list-group-flush">

      <div class="list-group-item px-1">
        <h6><a href="https://netbox.dev/blog/posts/netbox-v353-released/">NetBox v3.5.3 Released</a></h6>
        <div>
          NetBox v3.5.3 is now available!
        </div>
      </div>

      <div class="list-group-item px-1">
        <h6><a href="https://netbox.dev/blog/netdevops-days-2023-london-venue-date-initial-sponsors-speakers-announced-early-bird-tickets-now-live/">NetDevOps Days 2023 London - Venue, Date, Initial Sponsors &amp; Speakers Announced. Early Bird Tickets Now Live!</a></h6>
        <div>
          As momentum for NetDevOps Days 2023 London builds at breakneck speed, we're thrilled to bring you some exciting updates. Things are moving quickly, and the enthusiasm from the community is palpable!
        </div>
      </div>

      <div class="list-group-item px-1">
        <h6><a href="https://netbox.dev/blog/netdevops-days-2023-london-big-updates-and-early-bird-tickets-now-live/">NetDevOps Days 2023 London - Big updates and Early Bird Tickets Now Live!</a></h6>
        <div>
          As momentum for NetDevOps Days 2023 London builds at breakneck speed, we're thrilled to bring you some exciting updates!
        </div>
      </div>

      <div class="list-group-item px-1">
        <h6><a href="https://netbox.dev/blog/announcing-netdevops-days-2023/">Announcing NetDevOps Days 2023</a></h6>
        <div>
          Network automation is hotter than ever, but to date there haven't been any real practitioner focused events to bring the community together.
        </div>
      </div>

      <div class="list-group-item px-1">
        <h6><a href="https://netbox.dev/blog/posts/netbox-v352-released/">NetBox v3.5.2 Released</a></h6>
        <div>
          NetBox v3.5.2 is now available!
        </div>
      </div>

  </div>


    </div>
  </div>
</div>




<div
    class="grid-stack-item"
    gs-w="4"
    gs-h="3"
    gs-x="None"
    gs-y="None"
    gs-id="7829cedb-aa47-4300-8701-7820141c1d01"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-secondary p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/7829cedb-aa47-4300-8701-7820141c1d01/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/7829cedb-aa47-4300-8701-7820141c1d01/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>DCIM</strong>

    </div>
    <div class="card-body p-2 overflow-auto">

  <div class="list-group list-group-flush">


        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Sites
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Racks
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Device Types
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Devices
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Cables
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>


  </div>


    </div>
  </div>
</div>




<div
    class="grid-stack-item"
    gs-w="4"
    gs-h="2"
    gs-x="None"
    gs-y="None"
    gs-id="b2cd13d7-fcbb-4945-9159-06a7a9ad8fbd"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-secondary p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/b2cd13d7-fcbb-4945-9159-06a7a9ad8fbd/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/b2cd13d7-fcbb-4945-9159-06a7a9ad8fbd/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>Virtualization</strong>

    </div>
    <div class="card-body p-2 overflow-auto">

  <div class="list-group list-group-flush">


        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Clusters
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>



        <div class="list-group-item list-group-item-action d-flex w-100 justify-content-between align-items-center">
          Virtual Machines
          <h6 class="mb-1" title="No permission"><i class="mdi mdi-lock-outline"></i></h6>
        </div>


  </div>


    </div>
  </div>
</div>




<div
    class="grid-stack-item"
    gs-w="12"
    gs-h="5"
    gs-x="None"
    gs-y="None"
    gs-id="41013bfa-1288-416e-9031-88f97034d45d"
>
  <div class="card grid-stack-item-content">
    <div class="card-header text-center text-light bg-blue p-1">
      <div class="float-start ps-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/41013bfa-1288-416e-9031-88f97034d45d/configure/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-cog text-gray"></i></a>
      </div>
      <div class="float-end pe-1">
        <a href="#"
          hx-get="/extras/dashboard/widgets/41013bfa-1288-416e-9031-88f97034d45d/delete/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
        ><i class="mdi mdi-close text-gray"></i></a>
      </div>

        <strong>Change Log</strong>

    </div>
    <div class="card-body p-2 overflow-auto">

  <div class="text-muted text-center">
    <i class="mdi mdi-lock-outline"></i> No permission to view this content.
  </div>


    </div>
  </div>
</div>


  </div>
  <div class="collapse multi-collapse show">
    <div class="d-flex px-3">
      <div class="flex-grow-1">
        <button type="button" id="unlock_dashboard" class="btn btn-light btn-sm" data-bs-toggle="collapse" data-bs-target=".multi-collapse">
          <i class="mdi mdi-lock-open-outline"></i> Unlock Dashboard
        </button>
      </div>
    </div>
  </div>
  <div class="collapse multi-collapse">
    <div class="d-flex px-3">
      <div class="flex-grow-1">
        <button type="button" id="lock_dashboard" class="btn btn-light btn-sm" data-bs-toggle="collapse" data-bs-target=".multi-collapse">
          <i class="mdi mdi-lock-outline"></i> Lock Dashboard
        </button>
      </div>
      <div>
        <a href="#"
          hx-get="/extras/dashboard/widgets/add/"
          hx-target="#htmx-modal-content"
          data-bs-toggle="modal"
          data-bs-target="#htmx-modal"
          class="btn btn-success btn-sm"
        >
          <i class="mdi mdi-plus"></i> Add Widget
        </a>
        <button id="save_dashboard" class="btn btn-primary btn-sm" data-url="/api/extras/dashboard/">
          <i class="mdi mdi-content-save-outline"></i> Save Layout
        </button>
        <a href="/extras/dashboard/reset/" class="btn btn-danger btn-sm">
          <i class="mdi mdi-backspace"></i>  Reset Dashboard
        </a>
      </div>
    </div>
  </div>



        </div>



  <div class="modal fade" id="htmx-modal" tabindex="-1" aria-hidden="true">
  <div class="modal-dialog">
    <div class="modal-content" id="htmx-modal-content">

    </div>
  </div>
</div>




        <footer class="footer container-fluid">

            <div class="row align-items-center justify-content-between mx-0">

              <div class="col-sm-12 col-md-auto fs-4 noprint">
                <nav class="nav justify-content-center justify-content-lg-start">


                    <a type="button" class="nav-link" href="/static/docs/" target="_blank">
                      <i title="Docs" class="mdi mdi-book-open-variant text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>
                    </a>


                    <a type="button" class="nav-link" href="/api/" target="_blank">
                      <i title="REST API" class="mdi mdi-cloud-braces text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>
                    </a>


                    <a type="button" class="nav-link" href="/api/schema/swagger-ui/" target="_blank">
                      <i title="REST API documentation" class="mdi mdi-book text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>
                    </a>





                    <a type="button" class="nav-link" href="https://github.com/netbox-community/netbox" target="_blank">
                      <i title="Source Code" class="mdi mdi-github text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>
                    </a>


                    <a type="button" class="nav-link" href="https://netdev.chat/" target="_blank">
                      <i title="Community" class="mdi mdi-slack text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>
                    </a>

                </nav>
              </div>

              <div class="col-sm-12 col-md-auto text-center text-lg-end text-muted">
                <span class="d-block d-md-inline"><span title="June 11, 2023 8:30 a.m.">2023-06-11 08:30</span> UTC</span>
                <span class="ms-md-3 d-block d-md-inline">netbox (v3.5.3)</span>
              </div>

            </div>

        </footer>

      </div>

    </main>
  </div>








<div id="django-messages" class="toast-container">







</div>



    <div id="netbox-data" style="display: none!important; visibility: hidden!important">

    </div>

  </body>
</html>
* Connection #0 to host netbox.your.domain left intact

real    0m6.185s
user    0m0.212s
sys     0m0.026s

**Reply by chanukad24** [reply]
I just tried to run this command from outside and noticed port 443 is refused from the server. I opened port 443 from the firewall and allowed full access into ngix as well. Still same.

└─# time curl -v https://netbox.your.domain/
*   Trying X.X.X.X:443...
* connect to X.X.X.X port 443 failed: Connection refused
* Failed to connect to netbox.your.domain port 443 after 21170 ms: Connection refused
* Closing connection 0
curl: (7) Failed to connect to netbox.your.domain port 443 after 21170 ms: Connection refused

real    0m21.243s
user    0m0.016s
sys     0m0.016s



root@netbox:~# ufw status verbose
Status: active
Logging: on (low)
Default: deny (incoming), allow (outgoing), deny (routed)
New profiles: skip



To                         Action      From
--                         ------      ----
8001                       ALLOW IN    Anywhere
443                        ALLOW IN    Anywhere
5432                       ALLOW IN    Anywhere
80,443/tcp (Nginx Full)    ALLOW IN    Anywhere
22/tcp                     ALLOW IN    Anywhere
443/tcp                    ALLOW IN    Anywhere
8001 (v6)                  ALLOW IN    Anywhere (v6)
443 (v6)                   ALLOW IN    Anywhere (v6)
5432 (v6)                  ALLOW IN    Anywhere (v6)
80,443/tcp (Nginx Full (v6)) ALLOW IN    Anywhere (v6)
22/tcp (v6)                ALLOW IN    Anywhere (v6)
443/tcp (v6)               ALLOW IN    Anywhere (v6)
